common/goos/ParseHelpers.cpp

```diff
@@ -72,11 +77,15 @@ bool va_check(
     } else {
       // argument given.
       if (kv.second.second.has_value() && kv.second.second != kv2->second.type) {
-        // but is wrong type
-        *err_string = "Argument \"" + kv.first + "\" has type " +
-                      object_type_to_string(kv2->second.type) + " but " +
-                      object_type_to_string(kv.second.second.value()) + " was expected";
-        return false;
+        // special case -- an empty list is a valid pair
+        if (!(kv.second.second == goos::ObjectType::PAIR &&
```

Comment:
I think this will break error messages for a lot of things. An `EMPTY_LIST` in GOOS isn't actually a pair. So `foo.as_pair()` will fail on an empty list.

There's a lot of existing code that assumes that it's safe to call `.as_pair` if the `va_check` passes with the `::PAIR` type, and this change would break that assumption. I think it's better if functions that want either a pair or an empty string to manually check this.